src/Dofs/ConstraintHandler.jl

```diff
@@ -743,6 +714,53 @@ function apply!(KK::Union{SparseMatrixCSC, Symmetric}, f::AbstractVector, ch::Co
     return
 end
 
+"""
+    add_inhomogeneities!(f::AbstractVector, K::AbstractMatrix, ch::ConstraintHandler)
+
+Compute "f -= K*inhomogeneities".
+By default this is a generic version via SpMSpV kernel.
+"""
+function add_inhomogeneities!(f::AbstractVector, K::AbstractMatrix, inhomogeneities::AbstractVector, prescribed_dofs::AbstractVector{<:Integer}, dofmapping)
+    return f .-= K * sparsevec(prescribed_dofs, inhomogeneities, size(K, 2))
```

Comment:
(See combined fix below)
```suggestion
    return mul!(f, K, sparsevec(prescribed_dofs, inhomogeneities, size(K, 2)), -1, 1)
```
Reduces allocations. Significant relative speed diff for `SparseMatrixCSC`, but for `SparseMatrixCSR`, the call is dominated by a very slow sparse matrix sparse vector multiplication, which I guess should be fixed upstream.

Also needed to add `mul!` to using list from LinearAlgebra.